### ISSUE TITLE
Enforce superclasses of base rounds to always define required attributes

### ISSUE THREAD

**Issue by jmoreira-valory** [valory-xyz/open-autonomy#2246]
**Is your feature request related to a problem? Please describe.**
Some base rounds require a number of base attributes to be defined by the superclasses. For example, [none_type](https://github.com/valory-xyz/open-autonomy/blob/89a778cd71bbb56ed4805fa95470cd4f160217e7/packages/valory/skills/abstract_round_abci/base.py#L1548) on `CollectSameUntilThresholdRound`. 

If the superclass fails to initialize these attributes, the business logic will proceed without a warning. However, if during the execution variable is required to launch an event, it will [crash the application](https://github.com/valory-xyz/open-autonomy/blob/v0.14.14.post1/packages/valory/skills/abstract_round_abci/base.py#L1603-L1606) when accessing that attribute. 

**Describe the solution you'd like**
In summary, the application should throw a warning or exception when these required attributes are not initialized. This applies to several other base round classes.

**Describe alternatives you've considered**

**Additional context**

**Comment by Adamantios** [comment]
The solution in #2276 does not align with the intended one for this issue.
The agent crashes after it has started running, and only if the problematic round is entered.
Instead, we wanted to achieve a solution in which the check happens before the FSM cycle starts.
